File: share.py
import requests

# send(	file_name - Name of file to be sent from /music
# 	url [OPTIONAL] - URL to send the file to
# 	)
# returns:
#	ERROR_UPLOAD 100
#	ERROR_ALREADY_EXISTS 101
#	ERROR_NO_FILE_SENT 102
#	ERROR_MOVE_UPLOADED_FILE_FAILED 103
#	ERROR_NO_RESPONSE 160
#	SUCCESS 150

ERROR_NO_RESPONSE = 160
def send(file_name, user = "Anonymous", url = "http://localhost:8080/share"):
	files = { 'linkin_park' : open('music/' + file_name + '.wav', 'rb')}
	data = {'email': user}
	r = requests.post(url, files=files, data=data)
	if not r.text == '':
		return int(r.text)
	else:
		return ERROR_NO_RESPONSE

File: gae/share.php
<?php
	define ('SITE_ROOT', realpath(dirname(__FILE__)));
	$ERROR_UPLOAD = 100;
	$ERROR_ALREADY_EXISTS = 101;
	$ERROR_NO_FILE_SENT = 102;
	$ERROR_MOVE_UPLOADED_FILE_FAILED = 103;
	$SUCCESS = 150;

	if ($_FILES[linkin_park]["error"] == 0)
		if(count($_FILES) > 0)
			if (file_exists("upload/" . $_FILES[linkin_park]["name"]) == false)
				if(move_uploaded_file($_FILES[linkin_park]["tmp_name"],
					SITE_ROOT."/upload/".$_FILES[linkin_park]["name"]))
					echo $SUCCESS;
				else
					echo $ERROR_MOVE_UPLOADED_FILE_FAILED;
			else
				echo $ERROR_ALREADY_EXISTS;
		else
			echo $ERROR_NO_FILE_SENT;
	else
		echo $ERROR_UPLOAD." ".$_FILES[linkin_park]["error"];
?>
